sycl/test-e2e/README.md

```diff
@@ -385,56 +385,56 @@ project must keep using `<sycl/sycl.hpp>` provided by the SYCL2020
+  expanding for each selected device and without using the run_launcher.
 
 * `--param test-mode=build-only`
 
-  This mode can be used to compile all test binaries. To do this all
-  `UNSUPPORTED` and `REQUIRES` statements are ignored unless they contain
-  `UNSUPPORTED: true` or `REQUIRES: build-and-run-mode`. All `RUN:` lines within
-  a test are ran in this mode unless they contain the following expansions:
-  `%{run}`, `%{run-unfiltered-devices}`, or `%if run-mode`.
-
-  Currently, the only triple supported for `build-only` mode is `spir64`.
-
-  * `build-only` future work.
-
-    Note, the fact that `build-only` ignores general `UNSUPPORTED`/`REQUIRES`
-    statements is a current limitation. The logic for taking into account the
-    features that affect compilation, and ignoring those that are only relevant
-    to the execution of the program is currently being worked on.
-
-* `--param test-mode=run-only`
-
-  In this mode, tests will not be compiled, they will only run. To do this only
-  the `RUN:` lines that contain `%{run}`, `%{run-unfiltered-devices}` or `%if
-  run-mode` are executed. Tests are marked as unsupported in the same manner as
-  `full` mode. Since tests are not compiled in this mode, for any test to pass
-  the test binaries should already be in the `test_exec_root` directory, either
-  by having ran `full` or `build-only` modes previously on the system, or having
-  transferred the test binaries into that directory. The `run-mode` feature is
-  added when in this mode.
+  This mode can be used to compile all test binaries that can be built on the
+  system. To do this `REQUIRES`, and `UNSUPPORTED` statements are handled
+  differently to accommodate for the fact that in `build-only` mode we do not
+  have any devices, and as a result no device-specific features. Instead of
+  considering these features as missing, we assign a third "unknown" boolean
+  value to them. When evaluating An expressions it will result in an unknown
+  value if its result could be changed by setting the unknown features to either
+  true or false. i.e., `false || unknown = unknown` but `true || unknown = true`.
+  If an expression's final value is unknown we consider it to have met the
+  requirements. The list of device-agnostic features that are not considered
+  unknown in `build-only` is found in the `E2EExpr.py`.
+
+  The triples to compile for in this mode are set via the `sycl_build_targets`
+  lit parameter. Valid build targets are: `spir`,`nvidia`, `amd`, `native_cpu`.
+  These correspond to `spir64`, `nvptx64-nvidia-cuda`, `amdgcn-amd-amdhsa`, and
+  `native_cpu` triples respectively. Each build target should be separated with
+  a semicolon. This parameter is set to just `spir` by default. A test can be
+  marked as requiring, or not supporting a particular triple via the `target-*`
+  features. Note that we test if each build target is supported separately from
```

Comment:
Instead of "Note that we test ...", use something like "However, such an expression has additional constraint that each target availability should be detectable on its own without cross-target dependencies. The reason for this is that we are avoiding having to deal with boolean satisfiability problem when selecting targets to build for."